docs/proposals/cluster-deployment.md

```diff
@@ -0,0 +1,194 @@
+
+TODO: Enumerate ceritificates which have to be generated.
+
+### Step 3: Deploy master
+
+**Objective**: Run kubelet and all the required components (e.g. etcd, apiserver, scheduler, controllers) on the master machine.
+
+**Substeps**:
+
+1. copy certificates
+2. copy manifests for static pods:
+	1. etcd
+	2. apiserver, controller manager, scheduler
+3. run kubelet in docker container (configuration is read from apiserver Config object)
+4. run kubelet-checker in docker container
+
+**v1.2 simplifications**:
+
+1. kubelet-runner.sh - we will provide a custom docker image to run kubelet; it will contain
+kubelet binary and will run it using ```nsenter``` to workaround problem with mount propagation
+1. kubelet config file - we will read kubelet configuration file from disk instead of apiserver; it will
+be generated locally and copied to all nodes.
+
+**Exit criteria**:
+
+1. Can run basic API calls (e.g. create, list and delete pods) from the client side.
+2. Replication controller works
+
+### Step 4: Deploy nodes
+
+**Objective**: Run kubelet on all nodes.
+
+**Substeps**:
+
+1. copy certificates
+2. run kubelet in docker container (configuration is read from apiserver Config object)
+3. run kubelet-checker in docker container
+
+**v1.2 simplifications**:
+
+1. kubelet config file - we will read kubelet configuration file from disk instead of apiserver; it will
+be generated locally and copied to all nodes.
+
+**Exit critera**:
+
+1. All nodes are registered and ready.
+2. Can create a pod, which will be scheduled and run
+
+### Step 5: Setup kubernetes networking
+
+**Objective**: Configure the Kubernetes networking to allow routing requests to pods and services.
+
+To keep default setup consistent across open source deployments we will use Flannel to configure
+kubernetes networking. However, implementation of this step will allow to easily plug in different
+network solutions.
+
+**Substeps**:
+
+1. copy manifest for flannel server to master machine
+2. create a daemonset with flannel daemon (it will read assigned CIDR and configure network appropriately).
+
+**v1.2 simplifications**:
+
+1. flannel daemon will run as a standalone binary (not in docker container)
+2. flannel server will assign CIDRs to nodes outside of kubernetes; this will require restarting kubelet
+after reconfiguring network bridge on local machine; this will also require running master nad node differently
+(```--configure-cbr0=false``` on node and ```--allocate-node-cidrs=false``` on master), which breaks encapsulation
+between nodes
+
+**Exit criteria**:
+
+1. Pods correctly created, scheduled, run and accessible from all nodes.
+
+### Step 6: Add daemons
+
+**Objective:** Start all system daemons (e.g. kube-proxy)
+
+**Substeps:**:
+
+1. Create daemonset for kube-proxy
+
+**Exit criteria**:
+
+1. Services work correctly on all nodes.
+
+### Step 7: Add add-ons
+
+**Objective**: Add default add-ons (e.g. dns, logging)
```

Comment:
I'm not sure we should make logging a default add-on in this setup (since it's a bit deployment environment specific). You should add kube-ui here though.